### PR TITLE
LOOP-1100: Split out DeviceStatusIndicator from CGMManager protocol

### PR DESCRIPTION
**Pull request by rickpasetto**
Part of LOOP-1100:
- Split out `DeviceStatusIndicator` to its own protocol
- Added `filledImage` to `GlucoseTrend` extension

### CODE REVIEW

File: LoopKitUI/CGMManagerUI.swift [hunk 1]
```diff
@@ -19,7 +19,18 @@ public struct CGMManagerDescriptor {
     }
 }
 
-public protocol CGMManagerUI: CGMManager, DeviceManagerUI, PreferredGlucoseUnitObserver {
+public protocol DeviceStatusIndicator {
```
Comment (nhamming) [inline LoopKitUI/CGMManagerUI.swift]: include the status badge in here as well.

Comment (nhamming) [reply LoopKitUI/CGMManagerUI.swift]: Consider remaining to `CGMStatusIndicator`, since it isn't generic enough for any device.

**Review by nhamming — APPROVED** [review]
I don't want to block this, but the `CGMStatusBadge` needs to be put in the `DeviceStatusIndicator`. So approving with an understanding it will be moved. If this is a problem, let's chat about it.

LGTM

File: LoopKitUI/CGMManagerUI.swift (hunk 1 above)
Comment (ps2) [reply LoopKitUI/CGMManagerUI.swift]: I agree with @nhamming 's suggestion to name this more specifically.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**